waiter/src/waiter/scheduler.clj

```diff
@@ -149,6 +150,10 @@
      These options should be merged (to override) the global :deployment-error-config options
      in the waiter.state module when computing the deployment errors for the given service-id.")
 
+  (request-protocol [this instance port-index service-description]
```

Comment:
Is passing the `service-description` necessary, most scheduler instances can retrieve it using the service ID?

Reply:
You're probably right, but why do the lookup again when we already have the service-description handy everywhere that we want to call this function?

Reply:
That may be true today, but not in future call sites.

Reply:
You can make the same argument for `instance` vs `instance-id`. It seems unlikely that we'll be using this somewhere that we wouldn't also want to pull up the service-description for populating the request logs, right?

Reply:
We've refactored the signatures to remove the service description in the past, but we can do that later when needed if you are no onboard now.